src/main/java/org/terasology/logic/inventory/ItemCommands.java

```diff
@@ -114,6 +123,90 @@ public String give(
+    public String remove(
+            @Sender EntityRef client,
+            @CommandParam("prefabId") String itemPrefabName,
+            @CommandParam(value = "amount", required = false) Integer amount,
+            @CommandParam(value = "blockShapeName", required = false) String shapeUriParam) {
+
+        int itemAmount = amount != null ? amount : 1;
+        if (itemAmount < 1) {
+            return "Invalid amount of items!";
+        }
+
+        Set<ResourceUrn> matches = assetManager.resolve(itemPrefabName, Prefab.class);
+
+        if (matches.size() == 1) {
+            Prefab prefab = assetManager.getAsset(matches.iterator().next(), Prefab.class).orElse(null);
+
+            if (prefab != null && prefab.getComponent(ItemComponent.class) != null) {
+                EntityRef playerEntity = client.getComponent(ClientComponent.class).character;
+                List<EntityRef> itemsSlots = playerEntity.getComponent(InventoryComponent.class).itemSlots;
+                List<EntityRef> feasibleSlots = new ArrayList<>();
+                int removedItems = 0;
+                int tempItemAmount = itemAmount;
+
+                for (EntityRef slot : itemsSlots) {
+                    Prefab currentPrefab = slot.getParentPrefab();
+
+                    if (currentPrefab != null && currentPrefab.equals(prefab)) {
+                        feasibleSlots.add(slot);
+                        tempItemAmount = tempItemAmount - 1;
+                    }
+                }
+
+                if (tempItemAmount <= 0) {
+                    for (EntityRef slot: feasibleSlots) {
+                        EntityRef result = inventoryManager.removeItem(playerEntity, EntityRef.NULL, slot, true, 1);
+                        if (result == null) {
+                            return "Could not remove "
+                                    + prefab.getName();
+                        }
+                        if (result == EntityRef.NULL) {
+                            removedItems = removedItems + 1;
+                            if (removedItems == itemAmount) {
+                                break;
+                            }
+                        }
+                    }
+                }
+
+                if (removedItems > 0) {
+                    return "You removed "
+                            + (removedItems > 1 ? removedItems + " items of " : "an item of ")
+                            + prefab.getName()
+                            + (shapeUriParam != null ? " (Item can not have a shape)" : "");
+                } else {
+                    return "Could not find "
+                            + (itemAmount > 1 ? itemAmount + " items of " : "an item of ")
+                            + prefab.getName()
+                            + " in your inventory";
+                }
+            }
+        } else if (matches.size() > 1) {
+            StringBuilder builder = new StringBuilder();
```

Comment:
You mentioned that you wanted to try and reduce duplicate code in this file.
You could pull this section and the same in the give command into a private method like "displayItemDuplicates"